airbyte-integrations/connectors/source-snowflake/unit_tests/integration/test_table.py

```diff
@@ -171,6 +171,176 @@ def test_given_three_pages_read_then_return_records(self, uuid_mock, mock_auth,
+    @HttpMocker()
+    def test_retry_on_get_error(
+        self, 
+        error_code, 
+        property_mock, 
+        uuid_mock, 
+        mock_auth, 
+        http_mocker: 
+        HttpMocker) -> None:
+        _given_get_timezone(http_mocker)
+        _given_read_schema(http_mocker)
+        _given_table_catalog(http_mocker)
+        _given_table_with_primary_keys(http_mocker)
+
+        http_mocker.post(
+            table_request()
+            .with_table(_TABLE)
+            .with_requestID(_REQUESTID)
+            .with_async()
+            .build(),
+            snowflake_response("async_response",FieldPath("statementStatusUrl"))
+            .with_handle(_HANDLE)
+            .build()
+        )
+        http_mocker.get(
+            table_request()
+            .with_handle(_HANDLE)
+            .build(is_get=True),
+            [
+            snowflake_response("response_get_table",JsonPath("$.'data'"))
+            .with_record(a_snowflake_response("response_get_table",JsonPath("$.'data'[*]")))
+            .with_status_code(error_code).build(),
+            snowflake_response("response_get_table",JsonPath("$.'data'"))
+            .with_record(a_snowflake_response("response_get_table",JsonPath("$.'data'[*]"))).build(),
+            ]
+        )
+        output = _read(_config(), sync_mode=SyncMode.full_refresh)
+    
+    @parameterized.expand(
+            [
+                (400),
+                (429),
+                (random.randrange(400, 600)),
+                (random.randrange(400, 600)),
+                (random.randrange(400, 600)),
+            ],
+    )
+    @mock.patch("source_snowflake.streams.snowflake_parent_stream.SnowflakeStream.retry_factor", return_value=0, new_callable=mock.PropertyMock)
+    @mock.patch("source_snowflake.streams.snowflake_parent_stream.SnowflakeStream.max_retries", return_value=0, new_callable=mock.PropertyMock)
+    @mock.patch("source_snowflake.streams.snowflake_parent_stream.uuid.uuid4", return_value=_REQUESTID)
+    @mock.patch("source_snowflake.source.SnowflakeJwtAuthenticator")
+    @HttpMocker()
+    def test_fail_on_post_error_and_max_retry(
+        self,
+        error_code, 
+        mock_retries, 
+        mock_time, 
+        uuid_mock, 
+        mock_auth, 
+        http_mocker: HttpMocker) -> None:
+        _given_get_timezone(http_mocker)
+        _given_read_schema(http_mocker)
+        _given_table_catalog(http_mocker)
+        _given_table_with_primary_keys(http_mocker)
+
+        http_mocker.post(
+            table_request().with_table(_TABLE).with_requestID(_REQUESTID).with_async().build(),
+            snowflake_response("async_response",FieldPath("statementStatusUrl")).with_handle(_HANDLE).with_status_code(error_code).build()
+        )
+        output = _read(_config(), sync_mode=SyncMode.full_refresh, expecting_exception=True)
+        assert output.errors[-1].trace.error.failure_type == FailureType.config_error
```

Comment:
Why are 5XX HTTP errors config errors?

Reply:
@Ismail-Naziz i'll let you answer on ths one :)

Reply:
The default behavior I have put is to get a config error. which error are you expecting this case ? @maxi297

Reply:
I think the expected error for an error 500 is system faillure

Reply:
I mean, it all depends on the Snowflake API but usually, a 5XX HTTP status means that this is not a config issue